packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3804 @@
+                params: [
+                  '0x02e205010101825208946bf137f335ea1b8f193b8f6ea92561a60d23a2078080c0808080',
+                ],
+              },
+              response: {
+                result: '0x2',
+              },
+            },
+            // PendingTransactionTracker.#checkTransaction
+            {
+              request: {
+                method: 'eth_getTransactionReceipt',
+                params: ['0x1'],
+              },
+              response: {
+                result: null,
+              },
+            },
+            // BlockTracker
+            {
+              request: {
+                method: 'eth_blockNumber',
+                params: [],
+              },
+              response: {
+                result: '0x4',
+              },
+            },
+            // BlockTracker
+            {
+              request: {
+                method: 'eth_blockNumber',
+                params: [],
+              },
+              response: {
+                result: '0x4',
+              },
+            },
+            // PendingTransactionTracker.#checkTransaction
+            {
+              request: {
+                method: 'eth_getTransactionReceipt',
+                params: ['0x1'],
+              },
+              response: {
+                result: null,
+              },
+            },
+            // PendingTransactionTracker.#checkTransaction
+            {
+              request: {
+                method: 'eth_getTransactionReceipt',
+                params: ['0x2'],
+              },
+              response: {
+                result: {
+                  blockHash: '0x2',
+                  blockNumber: '0x4', // we need at least 2 blocks mocked since the first one is used for when the blockTracker is instantied before we have listeners
+                  status: '0x1', // 0x1 = success
+                },
+              },
+            },
+            // PendingTransactionTracker.#onTransactionConfirmed
+            {
+              request: {
+                method: 'eth_getBlockByHash',
+                params: ['0x2', false],
+              },
+              response: {
+                result: {
+                  transactions: [],
+                },
+              },
+            },
+          ],
+        });
+        const { transactionController, approvalController } =
+          await newController({});
+        const { result, transactionMeta } =
+          await transactionController.addTransaction(
+            {
+              from: ACCOUNT_MOCK,
+              to: ACCOUNT_2_MOCK,
+            },
+            { networkClientId: 'goerli' },
+          );
+
+        await approvalController.accept(transactionMeta.id);
+        await advanceTime({ clock, duration: 1 });
+
+        await result;
+
+        await transactionController.stopTransaction(transactionMeta.id);
+
+        // blocktracker polling is 20s
+        await advanceTime({ clock, duration: 20000 });
```

Comment:
What do you think about extracting this interval to a constant in this file so we can put a name to this piece of knowledge and don't have to leave a comment every time it appears?

Or, as an alternative, what do you think about extracting `await advanceTime({ clock, duration: 1 })` and `await advanceTime({ clock, duration: 20000 })` to well-named functions? The second one is clearly to wait for the block tracker, but the first one I'm confused by — is it just to flush promises, and if not, why give a time of 1ms as opposed to 0ms?

Reply:
For the blockTracker one: done here: https://github.com/MetaMask/core/pull/3885

> but the first one I'm confused by — is it just to flush promises, and if not, why give a time of 1ms as opposed to 0ms?

For this if you look at the `advanceTime` helper function:
```
export async function advanceTime({
  clock,
  duration,
  stepSize = duration / 4,
}: {
  clock: sinon.SinonFakeTimers;
  duration: number;
  stepSize?: number;
}): Promise<void> {
  do {
    await clock.tickAsync(stepSize);
    duration -= stepSize;
  } while (duration > 0);
}
```
Duration has to be greater than 0 in order for stepSize to be greater that 0 so that tick will happen. We could modify the helper so this isn't the case but that's how it is right now.

Reply:
Oh, right, I see. Gotcha.